### DISCUSSION TITLE
How to read private twitter which I follow

### DISCUSSION THREAD

**Discussion by MaciejGit** [redouane59/twittered#156] [Q&A]
Hi there,

I'm following few private accounts (restricted access, only approved users can read content and I'm one of these approved users, can read all content in the browser) and in my app using twittered library I would like to read content from these accounts. It used to work few months ago, but unfortunately it stopped at some point (I don't know when exactly) and now I'm getting the following error:

{"errors":[{"value":"123","parameter":"id","resource_type":"user","section":"data","title":"Authorization Error","detail":"Sorry, you are not authorized to see the user with id: [123].","type":"https://api.twitter.com/2/problems/not-authorized-for-resource"}]}

I have twittered app connected to my account, everything else is working. Any idea how to enable it again?

Thanks!

**Comment by redouane59** [comment]
Hi,

Can you tell me which version and which method you are using ? It can maybe be related to the V2 migration of some endpoints. 
If you could tell me what was the working version it will help me a lot debugging it :)

**Reply by MaciejGit** [reply]
Hello,

I tried master and develop (pulled yesterday) and I'm calling it from unit test: 

List<Tweet> result = twitterClient.getUserTimeline("123", 150);

Unfortunately I don't remember which version used to work previously, I only remember it was OK in November (more or less). I will try to pull different tags and let you know if I find something.

Thanks!

**Reply by redouane59** [reply]
Yes it would be great, just change the version on your `pom.xml`. 
This endpoint was migrate with `v1.15`

![image](https://user-images.githubusercontent.com/4500524/110386131-33404d80-8060-11eb-8c2a-ab8d86e4cd15.png)

So maybe there is something to fix with that.

**Reply by MaciejGit** [reply]
I tried v1.12 and it's working fine for getUserTimeline, however the text (twitter content) is truncated so I'm going to dig a bit deeper into it. 

But when I try:

Tweet getTweet(String tweetId) 

I'm getting similar error: 

{"errors":[{"value":"123","parameter":"id","resource_type":"tweet","section":"data","title":"Authorization Error","detail":"Sorry, you are not authorized to see the Tweet with id: [123].","type":"https://api.twitter.com/2/problems/not-authorized-for-resource"}]}

Are you going to look into it and fix it?

**Reply by redouane59** [reply]
Yes I will create a issue now and we will check it soon

**Reply by redouane59** [reply]
https://github.com/redouane59/twittered/pull/158 solved
